src/solvers/flattening/boolbv_index.cpp

```diff
@@ -258,7 +252,9 @@ bvt boolbvt::convert_index(const index_exprt &expr)
 
       typet constant_type=index.type(); // type of index operand
 
-      assert(array_size>0);
+      DATA_INVARIANT(
+        array_size > 0,
+        "non-positive array sizes are forbidden by the C standard");
```

Comment:
This isn't about C code anymore, so we should talk about goto programs, not C code.

Reply:
We wrote this error message because we thought that goto-programs essentially model C semantics. It seems our assumption is wrong.

Reply:
Will it be okay to change it to something like "non-positive array sizes are not allowed in goto-programs"? The only problem is that I also feel kind of uneasy not being able to point to a formal standard regarding semantics like that, no matter how self-evident or obvious it might seem. Thoughts @tautschnig ?

Reply:
Historically goto-programs and C semantics have been very close, but we are trying to get this disentangled (after all a considerable business is coming from the Java front-end :-) ). I understand your sentiment, and one day we'll hopefully get goto-program semantics documented. Please use the updated message that you suggested.